File: Rasha_Toine/Rasha/QLForm/src/ql/ast/expression/LiteralExpression.java
package ql.ast.expression;

import ql.visiting.ExpressionVisitor;
import ql.ast.literal.Literal;

public class LiteralExpression extends Expression {

	private final Literal literal;
	
	public LiteralExpression(Literal literal) {
		this.literal = literal;
	}

	@Override
	public <T, U> T accept(ExpressionVisitor<T, U> visitor, U ctx) {
		return visitor.visit(this, ctx);
	}

	public Literal getLiteral() {
		return literal;
	}
}

File: DogeQL/src/main/kotlin/doge/ast/node/expression/LiteralExpression.kt
package doge.ast.node.expression

import doge.ast.location.SourceLocation
import doge.data.value.BaseSymbolValue
import doge.visitor.QuestionnaireASTBaseVisitor

class LiteralExpression(val value: BaseSymbolValue, sourceLocation: SourceLocation) : Expression(sourceLocation) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}